pkg/util/deployer/deployer.go

```diff
@@ -0,0 +1,125 @@
+package deployer
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"io"
+	"io/fs"
+	"strings"
+	"text/template"
+
+	appsv1 "k8s.io/api/apps/v1"
+	kruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+
+	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
+	"github.com/Azure/ARO-RP/pkg/util/dynamichelper"
+	"github.com/Azure/ARO-RP/pkg/util/ready"
+)
+
+type Deployer interface {
+	CreateOrUpdate(context.Context, *arov1alpha1.Cluster, interface{}) error
+	Remove(context.Context, interface{}) error
+	IsReady(context.Context, string, string) (bool, error)
+	Template(interface{}, fs.FS) ([]kruntime.Object, error)
+}
+
+type deployer struct {
+	kubernetescli kubernetes.Interface
+	dh            dynamichelper.Interface
+	fs            fs.FS
+}
+
+func NewDeployer(kubernetescli kubernetes.Interface, dh dynamichelper.Interface, fs fs.FS) Deployer {
+	return &deployer{
+		kubernetescli: kubernetescli,
+		dh:            dh,
+		fs:            fs,
+	}
+}
+
+func (deployer *deployer) Template(data interface{}, fsys fs.FS) ([]kruntime.Object, error) {
+	results := make([]kruntime.Object, 0)
+	template, err := template.ParseFS(fsys, "staticresources/*")
+	if err != nil {
+		return nil, err
+	}
+
+	buffer := new(bytes.Buffer)
+	for _, v := range template.Templates() {
+		err := v.Execute(buffer, data)
```

Comment:
nit: what is `v`, and can it be renamed to something more meaningful?

Reply:
`v` stands for value. It is commonly used in range loops, alongside `k` for key. Don't mind renaming it though